### PR TITLE
Added the starting point for Identity

### PR DESCRIPTION
**Pull request by dave-cheong**
Also includes missing startDate for the relationshipType.controller payload serialization.

### CODE REVIEW

File: backend/typescript/models/identity.model.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        required: [true, 'Type is required'],
+        trim: true,
+        enum: IdentityType.values()
+    },
+    defaultValue: {
+        type: String
+    }
+});
+
+IdentitySchema.method('type', function () {
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: We need to update data model.
I got ok to change 'type' to identityType as type is keyword in many languages,

File: backend/typescript/controllers/relationshipType.controller.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export class RelationshipTypeController {
                     code: attributeNameUsage.attributeName.code,
                     shortDecodeText: attributeNameUsage.attributeName.shortDecodeText,
                     longDecodeText: attributeNameUsage.attributeName.longDecodeText,
+                    startDate: attributeNameUsage.attributeName.startDate,
```
Comment (sundriver) [inline backend/typescript/controllers/relationshipType.controller.ts]: how about endDate?

Comment (sundriver) [inline backend/typescript/controllers/relationshipType.controller.ts]: Considering the post by dloone, could you try to match the payload pattern

File: backend/typescript/models/identity.model.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    return IdentityType.valueOf(this.typeName);
+});
+
+/* tslint:disable:no-empty-interfaces */
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: Don't need this anymore

Comment (dave-cheong) [inline backend/typescript/models/identity.model.ts]: Removed unused tslint override.

File: backend/typescript/controllers/relationshipType.controller.ts (hunk 2 above)
Comment (dave-cheong) [inline backend/typescript/controllers/relationshipType.controller.ts]: I've added the endDate. But I'll address dloone's new api design in a subsequent pull request. So ignore for this.

File: backend/typescript/models/identity.model.ts (hunk 1 above)
Comment (dave-cheong) [inline backend/typescript/models/identity.model.ts]: Renamed to identityType

File: backend/typescript/models/identity.model.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    constructor(public name:String) {
+    }
+}
+
+export interface IIdentity extends IRAMObject {
+    idValue: string;
+    identityTypeName: string;
+    defaultValues: string;
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: data type of default value is object id

File: backend/typescript/models/identity.model.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        return null;
+    }
+
+    constructor(public name:String) {
+    }
+}
+
+export interface IIdentity extends IRAMObject {
+    idValue: string;
+    identityTypeName: string;
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: should be enum ?

File: backend/typescript/models/identity.model.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    defaultValues: string;
+    type(): IdentityType;
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: renaming this to identityType

File: backend/typescript/models/identity.model.ts (hunk 5 above)
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: identityTypename : "LinkId" | "PublicIdentifier" | ...

File: backend/typescript/models/identity.model.ts (hunk 4 above)
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: also it is defaultValue, without "s"

File: backend/typescript/models/identity.model.ts (hunk 5 above)
Comment (dave-cheong) [inline backend/typescript/models/identity.model.ts]: As agreed, we will stick with the pseudo enum (class) with enum validation on the schema.

**Pull request merged**